From 44b69e23fb7522e1e0d6052c4d379715f9aa6447 Mon Sep 17 00:00:00 2001
From: Michael Smith <michael.smith@puppet.com>
Date: Thu, 24 Aug 2017 16:43:01 -0700
Subject: [PATCH] Compilation fails on AIX with 7.55.1 (#1828)

Compilation fails on multi.c. The root of it is that on AIX, when
_ALL_SOURCE is set to 1 (as CURL does), sys/poll.h uses #define for
event and revent. Because multi.c includes sys/poll.h but multi.h
doesn't, we end up with a mismatch between names for the curl_waitfd
struct.

Previously this appears to have built with curl 7.51.0 because multi.h
also included sys/poll.h, so they were equally wrong. Based on
https://curl.haxx.se/mail/lib-2013-02/0330.html, this appears to have
been an issue off-and-on for years.

My workaround has been to undef _ALL_SOURCE before including select.h in
multi.c, and redefine it after. This probably isn't a good general
solution.
---
 lib/multi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/multi.c b/lib/multi.c
index d5bc532ea..ecc00d3af 100644
--- a/lib/multi.c
+++ b/lib/multi.c
@@ -35,7 +35,9 @@
 #include "sendf.h"
 #include "timeval.h"
 #include "http.h"
+#undef _ALL_SOURCE
 #include "select.h"
+#define _ALL_SOURCE 1
 #include "warnless.h"
 #include "speedcheck.h"
 #include "conncache.h"
-- 
2.14.1

